mod/state-transition/pkg/core/state_processor.go

```diff
@@ -219,6 +221,23 @@ func (sp *StateProcessor[
 			return nil, err
 		}
 
+		// Handle special cases
+		if sp.cs.DepositEth1ChainID() == spec.BoonetEth1ChainID &&
+			slot == math.U64(spec.BoonetFork2Height) {
+			var idx uint64
+			idx, err = st.GetEth1DepositIndex()
+			if err != nil {
+				return nil, fmt.Errorf(
+					"failed retrieving deposit index at slot %d: %w",
+					slot, err,
+				)
+			}
+			fixedDepositIdx := idx - 1
```

Comment:
no need to check whether idx>1 since we must have some validators (at least genesis ones)

Reply:
makes sense, can you put this as a comment on this line?